app/src/main/java/com/duckduckgo/app/di/DatabaseModule.kt

```diff
@@ -29,6 +36,23 @@ import javax.inject.Singleton
 @Module(includes = [DaoModule::class])
 class DatabaseModule {
 
+    @Provides
+    @Singleton
+    fun provideWebViewHttpAuthStore(
+        context: Context,
+        fireproofWebsiteDao: FireproofWebsiteDao
+    ): WebViewHttpAuthStore {
+        val httpAuthDb = if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
+            // This database is a Room wrapper around the http_auth convenience database
+            Room.databaseBuilder(context, HttpAuthDatabase::class.java, "http_auth.db")
+                .setJournalMode(RoomDatabase.JournalMode.TRUNCATE)
+                .addMigrations(HttpAuthDatabase.MIGRATION_1_2)
+                .build()
+        } else null
+
+        return RealWebViewHttpAuthStore(DefaultDispatcherProvider(), fireproofWebsiteDao, httpAuthDb?.httpAuthDao())
```

Comment:
Should we create this instance using an injected Dispatcher provider?

Reply:
It's s good question, I saw that all usages of `DefaultDispatcherProvider` are not through DI but rather building the type directly. That's why I didn't want to provide it, to avoid confusion.
As it is right now, it is still an injected dependency, just it is not available in the DI graph.
I think it is worth refactoring all places where we call `DefaultDispatcherProvider()` and get that thru DI but not sure that place is this PR.

Reply:
Agree, I don't think we need to do that refactor within this PR. 

However, we already have `CoroutinesModule` that will provide that `DispatcherProvider` for you. And we have some Modules using it already (`UncaughtExceptionModule`, `BrowserModule`,...) So I would say we can use it in the same way.

Reply:
oh nice, I missed the fact that we had it already in the `CoroutinesModule`. Will do